cypress/e2e/signIn.cy.js

```diff
@@ -2,6 +2,27 @@
 
 describe('Sign In page', () => {
   it('should provide an ability to log in', () => {
```

Comment:
This test should only check if there is a possibility to log in with valid credentials and if the username is displayed properly. 
You put way too many assertions and some of them are even duplicated. The only assertion you need in this task is: `cy.get('a.nav-link').should('contain.text', 'JohnSmith2010');`
We do not check registration or page composition here. Our precondition is that there already is a user created, so do not register a new one. Just use the credentials of the existing one. 
If you visit a login page directly the whole test may have 5-7 lines of code in total.